server/routes/sitemap.js

```diff
@@ -0,0 +1,141 @@
+import AWS from 'aws-sdk';
+import { SitemapAndIndexStream, SitemapStream } from 'sitemap';
+import * as stream from 'stream';
+import { promisify } from 'util';
+import { createGzip } from 'zlib';
+
+import { isProd } from 'utils/environment';
+import { Page, Pub } from 'server/models';
+import app from 'server/server';
+import { handleErrors } from 'server/utils/errors';
+import { getInitialData } from 'server/utils/initData';
+import { hostIsValid } from 'server/utils/routes';
+import { communityUrl, pubUrl, pageUrl } from 'utils/canonicalUrls';
+
+const s3 = new AWS.S3({ params: { Bucket: 'sitemaps.pubpub.org' } });
+const pipeline = promisify(stream.pipeline);
+const keyPrefix = isProd() ? 'prod' : 'dev';
+
+const uploadFromStream = (key) => {
+	const pass = new stream.PassThrough();
+	const params = { Key: key, Body: pass, ACL: 'public-read' };
```

Comment:
_(Hacker News guy with a flashlight up to his face)_ but when they went back to check the bucket a few days later...its ACL had set to `public-read-write`! 👻